hudi-common/src/main/java/org/apache/hudi/common/model/HoodieIndexesMetadata.java

```diff
@@ -76,19 +76,19 @@ public String toJson() throws JsonProcessingException {
    * Deserialize from JSON string to create an instance of this class.
    *
    * @param json Input JSON string.
-   * @return Deserialized instance of HoodieFunctionalIndexMetadata.
+   * @return Deserialized instance of HoodieIndexesMetadata.
    * @throws IOException If any deserialization errors occur.
    */
-  public static HoodieFunctionalIndexMetadata fromJson(String json) throws IOException {
+  public static HoodieIndexesMetadata fromJson(String json) throws IOException {
     if (json == null || json.isEmpty()) {
-      return new HoodieFunctionalIndexMetadata();
+      return new HoodieIndexesMetadata();
     }
-    return JsonUtils.getObjectMapper().readValue(json, HoodieFunctionalIndexMetadata.class);
+    return JsonUtils.getObjectMapper().readValue(json, HoodieIndexesMetadata.class);
   }
 
   @Override
   public String toString() {
-    return "HoodieFunctionalIndexMetadata{"
+    return "HoodieIndexesMetadata{"
```

Comment:
Let's avoid the hard code of the class name, we can fetch it through `getClass.getSimpleName`